src/server/scripts/EasternKingdoms/BlackrockMountain/BlackwingLair/boss_vaelastrasz.cpp

```diff
@@ -188,24 +191,19 @@ class boss_vaelastrasz : public CreatureScript
                         break;
                     case EVENT_BURNINGADRENALINE_CASTER:
                         {
-                            Unit* target = NULL;
-
-                            uint8 i = 0;
-                            while (i < 3)   // max 3 tries to get a random target with power_mana
+                            //selects a random target that isn't the current victim and is a mana user (selects mana users) but not pets
+                            //it also ignores targets who have the aura. We don't want to place the debuff on the same target twice.
+                            if (Unit *target = SelectTarget(SELECT_TARGET_RANDOM, 1, [&](Unit* u) { return !u->IsPet() && u && u->getPowerType() == POWER_MANA && !u->HasAura(SPELL_BURNINGADRENALINE); }))
```

Comment:
Added null check in Commit 192521b932f110341ac56ec5120affef55c6b1f7 but I do wonder if a null Unit will ever be provided to the PREDICATE. I've no experience in C++ but should I just be nullchecking any points TC spits out?

Also, GetTarget() says it won't ever be null. Does not null checking that cause static analysis issues?